src/caches.jl

```diff
@@ -113,14 +123,19 @@ function OrdinaryDiffEq.alg_cache(
 
     # Measurement model related things
     R = zeros(uElType, d, d)
-    H = f isa DynamicalODEFunction ? copy(E2) : copy(E1)
+    H_base = if f isa DynamicalODEFunction
+        copy(E2)
+    else
+        f.mass_matrix * E1
+    end
+    H = copy(H_base)
     v = zeros(uElType, d)
     S = PSDMatrix(zeros(uElType, D, d))
     measurement = Gaussian(v, S)
 
     # Caches
     du = f isa DynamicalODEFunction ? similar(u[2, :]) : similar(u)
-    ddu = f isa DynamicalODEFunction ? zeros(uElType, d, 2d) : zeros(uElType, d, d)
+    ddu = zeros(uElType, length(u), length(u))
```

Comment:
This is because for SecondOrderODEProblems we now compute jacobians of `f`, not `f.f1`. That's not the most efficient. But at the same time, using `OrdinaryDiffEq.calc_J!` instead of some self-written stuff is nice, so maybe that's overall a plus.

Reply:
Not an issue for now